### PR TITLE
Add exception when both region and custom instance metadata is config…

### PR DESCRIPTION
**Pull request by neha-bhargava**
Fixes # 
#3014 

**Changes proposed in this request**
Add exception when both region and custom metadata are configured

**Testing**
Added integration test

**Performance impact**
None

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/ConfidentialClientApplication.cs [hunk 1]
```diff
@@ -44,6 +44,11 @@ public sealed partial class ConfidentialClientApplication
         {
             GuardMobileFrameworks();
 
+            if (!(string.IsNullOrEmpty(configuration.AzureRegion) || configuration.CustomInstanceDiscoveryMetadata == null))
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/ConfidentialClientApplication.cs:47]: Try to use the Validate method on the CCABuilder class instead for this check.

Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/ConfidentialClientApplication.cs:47]: Isn't there an overload to custom instance metadata with URI?

**Review by bgavrilMS — APPROVED** [review]
Small changes needed.

File: src/client/Microsoft.Identity.Client/ConfidentialClientApplication.cs (hunk 1 above)
Comment (neha-bhargava) [reply src/client/Microsoft.Identity.Client/ConfidentialClientApplication.cs]: Yeah.. included it.

**Review by trwalke — APPROVED** [review]
Looks good, Thanks @neha-bhargava

**Review by pmaytak — APPROVED** [review]

**Pull request merged**